x/ccv/provider/keeper/throttle.go

```diff
@@ -29,9 +29,25 @@ func (k Keeper) HandlePendingSlashPackets(ctx sdktypes.Context) {
 	// Iterate through ordered (by received time) slash packet entries from any consumer chain
 	k.IteratePendingSlashPacketEntries(ctx, func(entry providertypes.SlashPacketEntry) (stop bool) {
 
+		// Obtain validator from consensus address.
+		// The slash packet validator address may be known only on the consumer chain
+		// in this case, it must be mapped back to the consensus address on the provider chain
+		consumerConsAddr := sdktypes.ConsAddress(entry.ValAddr)
+		providerConsAddr := k.GetProviderAddrFromConsumerAddr(ctx, entry.ConsumerChainID, consumerConsAddr)
+
+		// Note: if validator is not found or unbonded, this will be handled appropriately in HandleSlashPacket
+		val, found := k.stakingKeeper.GetValidatorByConsAddr(ctx, providerConsAddr)
+
 		// Obtain the validator power relevant to the slash packet that's about to be handled
 		// (this power will be removed via jailing or tombstoning)
-		valPower := k.stakingKeeper.GetLastValidatorPower(ctx, entry.ValAddr)
+		var valPower int64
+		if val.IsJailed() || !found {
```

Comment:
If `found==false` and `val==nil`, wouldn't this code panic?

Reply:
`val` is a `stakingtypes.Validator` concrete type, so it can never be nil. If `found` is `false`, then a zero value [is returned](https://github.com/cosmos/cosmos-sdk/blob/9c145c827001222df2e3e1101010874aeac20997/x/staking/keeper/validator.go#L41) for the validator type. 


If val was a pointer to a `stakingtypes.Validator` then this code could panic. However that'd require a breaking change to the staking keeper API

Reply:
Although, swapping the order as I've done [here](https://github.com/cosmos/interchain-security/pull/565/commits/74441c2d56a13a570a975526a2b1d155bf342cbb) will allow us not to rely on the value of `val` when `found = false`